timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosResourcesFactory.java

```diff
@@ -53,10 +58,52 @@ public static PaxosResources create(
         PaxosUseCaseContext timestampContext =
                 timestampContext(install, metrics, paxosRuntime, sharedExecutor, remoteClients);
 
-        return ImmutablePaxosResources.builder()
+        ImmutablePaxosResources.Builder resourcesBuilder = ImmutablePaxosResources.builder()
                 .timestamp(timestampContext)
-                .addAdhocResources(new TimestampPaxosResource(timestampContext.components()))
-                .build();
+                .addAdhocResources(new TimestampPaxosResource(timestampContext.components()));
+
+        if (install.useLeaderForEachClient()) {
+            throw new UnsupportedOperationException("not implemented yet");
+        } else {
+            TimelockPaxosMetrics timelockMetrics =
+                    TimelockPaxosMetrics.of(PaxosUseCase.LEADER_FOR_ALL_CLIENTS, metrics.getTaggedRegistry());
+
+            Factories.LeaderPingerFactory leaderPingerFactory = dependencies -> new SingleLeaderPinger(
+                    Maps.toMap(
+                            dependencies.remoteClients().nonBatchPingableLeaders(),
+                            _pingableLeader -> dependencies.sharedExecutor()),
+                    dependencies.leaderPingResponseWait(),
+                    dependencies.leaderUuid());
+
+            Factories.LeaderPingHealthCheckFactory healthCheckFactory = dependencies -> {
+                Set<PingableLeader> allPingableLeaders = ImmutableSet.<PingableLeader>builder()
+                        .add(dependencies.components().pingableLeader(PaxosUseCase.PSEUDO_LEADERSHIP_CLIENT))
+                        .addAll(dependencies.remoteClients().nonBatchPingableLeaders())
+                        .build();
+                return new LeaderPingHealthCheck(allPingableLeaders);
+            };
+
+            LeadershipContextFactory factory = ImmutableLeadershipContextFactory.builder()
+                    .install(install)
+                    .sharedExecutor(sharedExecutor)
+                    .remoteClients(remoteClients)
+                    .runtime(paxosRuntime)
+                    .useCase(PaxosUseCase.LEADER_FOR_ALL_CLIENTS)
+                    .metrics(timelockMetrics)
+                    .networkClientFactoryBuilder(ImmutableSingleLeaderNetworkClientFactories.builder())
+                    .leaderPingerFactory(leaderPingerFactory)
+                    .leaderPingHealthCheckFactory(healthCheckFactory)
+                    .build();
+
+            resourcesBuilder.leadershipContextFactory(factory);
+            resourcesBuilder.addAdhocResources(
+                    new LeadershipResource(
+                            factory.components().acceptor(PaxosUseCase.PSEUDO_LEADERSHIP_CLIENT),
+                            factory.components().learner(PaxosUseCase.PSEUDO_LEADERSHIP_CLIENT)),
+                    factory.components().pingableLeader(PaxosUseCase.PSEUDO_LEADERSHIP_CLIENT));
+        }
```

Comment:
This part looks right, but we should see if it's possible to split this up into functions etc. so it's easier to read

Reply:
I've extracted logic for single leader into a `configureLeaderForAllClients` method. It's quite self contained there and I don't think it makes sense to break it even further.